File: Sources/rust/src/file_utils.rs
use std::fs;
use std::path::Path;

pub fn list_files(path: &Path, extension: &str) -> Vec<String> {
    let mut files = Vec::new();
    if path.is_dir() {
        if let Ok(entries) = fs::read_dir(path) {
            for entry in entries.flatten() {
                let path = entry.path();
                
                if path.is_file() && path.extension().unwrap_or_default() == extension {
                    if let Some(str_path) = path.to_str() {
                        files.push(str_path.to_owned());
                    }
                }
            }
        }
    }
    files
}

File: Sources/cpp/src/file_utils.cpp
#include <filesystem>
#include <iostream>
#include <string>
#include <vector>

#include "file_utils.h"

namespace fs = std::filesystem;

std::vector<std::string> listFiles(const std::string& directory, const std::string& extension) {
    std::vector<std::string> files;
    for (const auto& entry : fs::directory_iterator(directory)) {
        if (entry.is_regular_file()) {
            auto& path = entry.path();
            if (path.extension() == extension) {
                files.push_back(path.string());
            }
        }
    }
    return files;
}

std::string fileName(const std::string& path) {
    std::filesystem::path fs_path(path);
    return fs_path.stem().string(); 
}